### PR TITLE
Feature 121/invitation section

### PR DESCRIPTION
**Pull request by tatianaKonovalova**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/store/invitations/mutations.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    //   apartment: {
+    //     ...payload.apartment,
+    //     address: payload.houseAddress,
+    //   },
+    //   email: payload.email,
+    //   id: payload.id,
+    //   status: payload.status,
+    //   type: payload.type,
+    // };
+    // state.invitations?.push(invitation);
```
Comment (aleonovdementiev) [inline src/store/invitations/mutations.ts]: Do you need this commented code here?

File: src/components/CreateInvitationForm.vue [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+      <label class="dialog-item" for="invitation_type">Тип запрошення : </label>
+      <Dropdown
+        id="invitation_type"
+        v-model="selectedData.selectedType"
+        :options="invitationData.invitationType"
+        placeholder="Оберіть тип запрошення"
+        :class="{
+          'p-invalid': v$.selectedData.selectedType.$error,
+        }"
+        @blur="v$.selectedData.selectedType.$touch"
```
Comment (tsehla42) [inline src/components/CreateInvitationForm.vue:13]: Please do not mix static attributes with dynamic and directives

v-model
:attribute1=
:attribute2=
arrtibute3=
...
@event=

File: src/components/CreateInvitationForm.vue [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    <p>
+      <label class="dialog-item dialog-item-address" for="user_email">Email : </label>
+      <InputText
+        id="user_email"
+        placeholder="Email"
+        v-model.trim="invitationData.email"
+        :class="{
+          'p-invalid': v$.invitationData.email.$error,
+        }"
+        @blur="v$.invitationData.email.$touch"
```
Comment (tsehla42) [inline src/components/CreateInvitationForm.vue:28]: Mixed attributes

File: src/components/CreateInvitationForm.vue [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    <div class="buttons-container">
+      <Button
+        label="Відправити"
+        icon="pi pi-check"
+        @click="createInvitation"
+        autofocus
+        class="p-button-info"
+        type="button"
+        value="Submit"
+        :disabled="v$.$invalid"
```
Comment (tsehla42) [inline src/components/CreateInvitationForm.vue:84]: Vue convention suggests to put :v-bind events at the very bottom

File: src/components/CreateInvitationForm.vue [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+      />
+      <Button
+        label="Скасувати"
+        icon="pi pi-times"
+        @click="closeCreatingInvitModel"
+        class="p-button-outlined p-button-info"
```
Comment (tsehla42) [inline src/components/CreateInvitationForm.vue:90]: Same here

File: src/components/CreateInvitationForm.vue [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+        selectedType: { requiredValidator },
+        selectedApartment: { requiredValidator },
+      },
+    };
+  },
+  async mounted(): Promise<void> {
+    try {
+      await this.$store.dispatch(`${StoreModuleEnum.housesStore}/${HousesActionsEnum.SET_HOUSES}`, this.cooperationId);
+    } catch {
+      console.log('error was caught during mounting');
```
Comment (tsehla42) [inline src/components/CreateInvitationForm.vue:163]: Consider using toast notification here

File: src/components/CreateInvitationForm.vue [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+    }
+  },
+  methods: {
+    closeCreatingInvitModel(): void {
```
Comment (tsehla42) [inline src/components/CreateInvitationForm.vue:167]: Fix typos => closeCreatingInviteModal

File: src/components/CreateInvitationForm.vue [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+      this.v$.$reset();
+      this.resetHouseDataFields(this.selectedData);
+      this.$emit('close-invit-model');
```
Comment (tsehla42) [inline src/components/CreateInvitationForm.vue:170]: Fix typos

File: src/components/CreateInvitationForm.vue [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+        role: 'user',
+      }; // which inteface the payload has? (disadvantage)
+
+      // await this.$store.dispatch(
+      //   `${StoreModuleEnum.invitationsStore}/${InvitationsActionsEnum.CREATE_INVITATION}`,
+      //   payload
+      // );
+
+      // Or to dispatch action which gets all invitations (we have an error due to not being able to access fields (street, houseNumber and so on))
+      // but it works
```
Comment (aleonovdementiev) [inline src/components/CreateInvitationForm.vue]: Remove commented code pls

File: src/components/CreateInvitationForm.vue [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+      this.resetHouseDataFields(this.selectedData);
+      this.$emit('close-invit-model');
+    },
+    resetHouseDataFields(data: any): void {
+      this.invitationData.email = '';
+      for (let field in data) {
+        data[field] = '';
+      }
+    },
+    async onChangeHouse(houseId: number, houseData: any): Promise<void> {
```
Comment (aleonovdementiev) [inline src/components/CreateInvitationForm.vue]: Do you need houseDate argument in this function?

File: src/store/invitations/actions.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+      const payloadToSend = new PostInvitationDTOModel(payload);
+      const { data } = await HTTP.post(`/invitations/create-invitation`, payloadToSend);
+      // const receivedData = new InvitationModel(data);
+
+      // const newInvintation = {
+      //   ...receivedData,
+      //   ...payload[1],
+      // };
+
+      // commit(InvitationsMutationsEnum.CREATE_INVITATION, newInvintation);
```
Comment (aleonovdementiev) [inline src/store/invitations/actions.ts]: Remove commented code pls

File: src/store/invitations/models/invitationApart.model.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import { AddressModel } from '@/shared/models/address.model';
+import { InvitationApartDTOModel } from '@/store/invitations/models/invitationApartDTO.model';
+
+export class InvitationApartModel {
```
Comment (aleonovdementiev) [inline src/store/invitations/models/invitationApart.model.ts]: Please, do not use shorthands in places, where you can avoid them) 'InvitationApartmentModel' is not a long name)

File: src/store/invitations/models/invitationApartDTO.model.ts [hunk 13]
```diff
@@ -0,0 +1,16 @@
+import { AddressDTOModel } from '@/shared/models/addressDTO.model';
+import { InvitationApartModel } from '@/store/invitations/models/invitationApart.model';
+
+export class InvitationApartDTOModel {
```
Comment (aleonovdementiev) [inline src/store/invitations/models/invitationApartDTO.model.ts]: Same here

**Review by akononenko2375 — APPROVED** [review]

**Review by aleonovdementiev — APPROVED** [review]

**Pull request merged**